cpp/order/Steinhardt.cc

```diff
@@ -184,80 +234,117 @@ void Steinhardt::computeAve(const freud::locality::NeighborList* nlist,
                 for (freud::locality::NeighborBond nb2 = ns_neighbors_iter->next(); !ns_neighbors_iter->end();
                      nb2 = ns_neighbors_iter->next())
                 {
-                    for (unsigned int k = 0; k < m_num_ms; ++k)
+                    for (size_t l_index = 0; l_index < m_ls.size(); ++l_index)
                     {
-                        // Adding all the qlm of the neighbors. We use the
-                        // vector function signature for indexing into the
-                        // arrays for speed.
-                        m_qlmiAve({static_cast<unsigned int>(i), k}) += m_qlmi({nb2.point_idx, k});
+                        auto& qlmiAve = m_qlmiAve[l_index];
+                        auto& qlmi = m_qlmi[l_index];
+                        const auto ave_index = qlmiAve.getIndex({static_cast<unsigned int>(i), 0});
+                        const auto nb2_index = qlmi.getIndex({nb2.point_idx, 0});
+                        for (unsigned int k = 0; k < m_num_ms[l_index]; ++k)
```

Comment:
Double-check that we use `size_t` for indexing.
```suggestion
                        for (size_t k = 0; k < m_num_ms[l_index]; ++k)
```

Reply:
We do, I think I got confused with something converting to `unsigned int` rather than `size_t`.